service/internal/credentials/vault.go

```diff
@@ -463,15 +468,16 @@ func (v VaultProvider) readRoleID(appRoleName string) (string, error) {
 	return secret.Data["role_id"].(string), nil
 }
 
-func (v VaultProvider) generateSecrets(appRoleName string) (string, string, error) {
+func (v VaultProvider) generateSecrets(appRoleName string) (*vault.Secret, error) {
```

Comment:
Since this method is private and used in a Vault context, figured it was safe to return a `vault.Secret` type.